docs/DeveloperGuide.adoc

```diff
@@ -111,156 +111,355 @@ image::DeleteSequenceDiagram.png[]
+If the user provides difficulty range as one of the search restrictions, `FindCommandParser` expects the format `LOWER_BOUND \<= difficulty \<= UPPER_BOUND` while `LOWER_BOUND` and `UPPER_BOUND` are valid strings for doubles (i.e. parsable by `Double.parseDouble(...)`).
+
+* `FindCommand`
++
+It creates and stores the `predicate` from an instance of `FindProblemDescriptor`. `predicate` is used to perform the filtering of the displayed problem list when the command is executed.
++
+`predicate` returns true only when the provided problem fulfills all restrictions described by the provided instance of `FindProblemDescriptor`.
++
+Additionally, it implements the following operation(s):
++
+** `FindCommand#execute(...)` - This method overrides `Command#execute(...)`. It filters problems in `filteredProblemList` in `model` with `predicate`.
+** `FindCommand#equals(...)` - Two instances of `FindCommand` are equal if and only if their `predicate` are equal.
+
+image::find/FindCommandClassDiagram.png[]
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+_Figure 2. Class Diagram of the Find Feature_
 
-image::UndoRedoState4.png[]
+Given below is an example usage scenario and how the find problem mechanism behaves at each step.
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+Step 1. The user executes `find t/recursion diff/2.0-4.0` to find a problem with a tag “recursion” and difficulty between 2.0 and 4.0.
 
-image::UndoRedoState5.png[]
+Step 2. `FindCommandParser` processes the user input and returns a `FindCommand` instance with the information of user’s search restrictions.
+
+[NOTE]
+If no valid search restriction is provided by the user, `FindCommandParser` will throw a parsing exception, which is handled and displayed to the user.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+Step 3. `LogicManager` invokes `execute()` method of the returned `FindCommand`. `FindCommand` updates the problem list with user’s search restrictions.
 
-image::CommitActivityDiagram.png[]
+image::find/FindCommandSequenceDiagram.png[]
+
+_Figure 3. Sequence Diagram for the Execution of `find` Command_
```

Comment:
Should your "updateFilteredProblemList" function call create a small activation bar under the Model class?